### PR TITLE
Allow site select to be overridden

### PR DESCRIPTION
**Pull request by mmcachran**
Allow wp_get_sites() args and select site dropdown to be overridden

### CODE REVIEW

File: include/admin.php [hunk 1]
```diff
@@ -133,7 +133,7 @@ public static function network_page_admin_duplicate_site() {
             // Load template if at least one Site is available
             if( $site_list ) {
 
-                $select_site_list = MUCD_Admin::select_site_list($site_list, $data['source']);
+                $select_site_list = apply_filters( 'mucd_select_site_list', MUCD_Admin::select_site_list( $site_list, $data['source'] ) );
```
Comment (jtsternberg) [inline include/admin.php]: In this case, the `MUCD_Admin::select_site_list` query will still run. We need a way to bypass that. Might look at something like this to instead: https://github.com/WebDevStudios/CMB2/blob/master/includes/CMB2_Field.php#L280

**Comment by mmcachran** [comment]
@jtsternberg that's a good point.  Even though the args for wp_get_sites() could be overridden to return only one site, the select was still being built unnecessarily.  Moving the override to the select_site_list() method and returning early will prevent the select from being built.

**Comment by mmcachran** [comment]
The first new filter is 'mucd_get_site_list_args' and is used to modify the arguments getting passed to wp_get_sites.  This is necessary to override the limit on bigger networks and avoid performance issues.  The second new filter is 'mucd_override_site_select' and is used to override the default site select and allow custom AJAX solutions to be implemented.

**Comment by jtsternberg** [comment]
To be clear, we're trying to short-circuit issues that can occur on a site with 10s of thousands of sites. These filters, as @mmcachran said, allow us replace the solution with something more performant. We can also submit a PR to replace with our solution completely if you prefer. (We'll use select2 and an ajax solution for selecting a site, vs building a dropdown).

**Comment by pierre-dargham** [comment]
**Hi there !**

I'm glad you opened this pull request, because several users of the plugin complained that selecting the site within a large amount (> 100) was difficult.

First of all, we just had time to increase the MUCD_MAX_NUMBER_OF_SITE from 1000 to 5000, and as you see, I even forgot to add a filter on it.

Then we added this issue on our roadmap for next version (1.4.0) and I wanted to provide an ajax solution with text-box and auto-suggestion. I'm not sure yet if we want to have this new selecting method for all cases, or just for cases with more than a certain amount of sites (50, 100 ?).

But you came with your pull request before I even had time to start working on this issue, and I think your work could be totally useful !

**What I suggest is to add those two changes to the plugin  :**

1. Adding filters on site selector and on site list, just as your pull request provides it (I will run some tests and merge things within a few days)

2. Replace the default select-box with an ajax text-box with auto-suggestion. If it is what you did, as @jtsternberg said, could you open a second pull request with your solution ?

I'm a little overwhelmed right now, but I will try to release version 1.4.0 with those two things within a month.

**Thank you for your help !** (and sorry for my bad english, I'm tired !)

**Best regards,**

Pierre

**Comment by jtsternberg** [comment]
@pierre-dargham glad to help! We'll be working on the ajax solution (implementing [select2](http://select2.github.io/select2/) in the next few days, so once we get a workable solution, we'll be happy to submit a pull request. In answer to your question, I would think it should always use the ajax solution as the UI is generally better than a dropdown, and wouldn't confuse the user if the UI suddenly changed because they added another site and it crossed the threshold.

**Comment by jtsternberg** [comment]
If you want, it might make more sense to disregard this pull request, or wait till we submit our changes since things might change.

**Comment by pierre-dargham** [comment]
I will leave it open for now, but I won't merge anything before I could see your second pull request with select2 :)

I think having both default selecting method with select2 / ajax AND filters on site list / select box is the best thing to do.

**Pull request closed without merging**